apps/desktop/src/components/Onboarding/connectOrCreate/useOnboardWithSocial.tsx

```diff
@@ -24,13 +24,13 @@ export const useOnboardWithSocial = (idp: Auth.IDP, onAuth: () => void) => {
     () =>
       handleAsyncAction(
         async () => {
-          const { secretKey, name } = await withTimeout(
+          const { secretKey, name, id } = await withTimeout(
             () => Auth.forIDP(idp, "desktop").getCredentials(),
             LOGIN_TIMEOUT
           );
           const { pk, pkh } = await getPublicKeyPairFromSk(secretKey);
-          restoreSocial(pk, pkh, name, idp);
-          toast({ description: `Successfully added ${name} account`, status: "success" });
+          restoreSocial(pk, pkh, name || id, idp);
```

Comment:
here it should be `email || name || id` to preserve [the same logic](https://github.com/trilitech/umami-v2/pull/1626/files#diff-159e8554a760382f6ca9a2bcb9f057d168ccb433f37c7c2efaee615564fd7289L59)

Reply:
this please
> The other option could be to keep returning const userData = { pk, pkh, typeOfLogin: loginType, id, name, emailAddress: email, imageUrl }; to dapps that use umami embed
But on all umami surfaces still use email || name || id as a name (e.g. desktop, embed modals)

Reply:
Fixed now, thank you!